lib/exd/api/crud.ex

```diff
@@ -213,8 +221,22 @@ defmodule Exd.Api.Crud do
   end
   defp check_error(_, _), do: nil
 
+  defp format_select_response([]), do: []
+  defp format_select_response([row | data]) when is_tuple(row) do
+    {:done, Enum.map(data, fn(r) -> Enum.map((r |> Tuple.to_list), fn(rec) -> Map.to_list(rec) end) |> :lists.flatten end)}
+  end
+  defp format_select_response([row | data]) when is_map(row) do
+    [row | data]
+  end
+
   defp format_data(data, api, opts) do
-    unless_error(data, api, export_data(data, opts))
+    # 'select' response is not __struct__, so we need to  format in other place
+    case format_select_response(data) do
```

Comment:
What happens, if you doesn't get list? (try get ... id:1) What happens, if database is unreachable, try to stop mariadb and make a query?